scripts/deposit.js

```diff
@@ -0,0 +1,24 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable indent */
+const SnappBase = artifacts.require("SnappBase")
+
+const zero_address = "0x0000000000000000000000000000000000000000"
+const one_hash = "0x0000000000000000000000000000000000000001"
+
+module.exports = async (callback) => {
+    const instance = await SnappBase.deployed()
+    const [accountId, tokenId, amount] = await process.argv.slice(4)
+
+    const depositor = await instance.accountToPublicKeyMap.call(accountId)
+    if (depositor == zero_address) {
+        console.log("No account registerd at index %s", accountId)
+        callback()
+    }
+    const slot = (await instance.depositSlot.call()).toNumber()
+    const tx = await instance.deposit(tokenId, amount, {from: depositor})
```

Comment:
It would be better to get the slot number from the event. The way you have done it, it will only be correct in 99 percent of all cases on a public testnet

Reply:
Yes, but these scripts aren't meant to listen to events... Oh! You mean I can get the event from the transaction?!

Reply:
Yes I will absolutely do this.